phonopy/structure/cells.py

```diff
@@ -290,21 +290,25 @@ def _primitive_cell(self, supercell):
             raise ValueError
 
     def _supercell_to_primitive_map(self, pos):
-        frac_pos = np.dot(pos, np.linalg.inv(self._primitive_matrix))
+        frac_pos = np.dot(pos, np.linalg.inv(self._primitive_matrix).T)
 
         p2s_positions = frac_pos[self._p2s_map]
         s2p_map = []
         for s_pos in frac_pos:
 
-            diffs = p2s_positions - s_pos
-            diffs -= np.rint(diffs)
+            # Compute distances from s_pos to all positions in _p2s_map.
+            frac_diffs = p2s_positions - s_pos
+            frac_diffs -= np.rint(frac_diffs)
 
-            matches = (abs(diffs) < self._symprec).all(axis=1).tolist()
+            cart_diffs = np.dot(frac_diffs, self._primitive_matrix.T)
```

Comment:
Interestingly, deleting this `.T` does not break any test cases despite being clearly wrong.  This indicates there is a hole in test coverage; I'll try to come up with a test for it.

Reply:
Update: After looking into this I'm only *more* confounded.  There are already three tests where `_primitive_matrix` is asymmetric, and they have nontrivial-looking positions; yet none of them break when I make this change.

I can't believe that all three of these cases just miraculously succeed by coincidence.

Reply:
Never mind, this is not too unusual.  In the best case scenario where a difference is exactly zero, multiplying it by *any* matrix will still produce zero.  Multiplying by an incorrect matrix only effects our error estimate for vectors that are "almost zero."

(this while deleting the `.T` is wrong here, it is not surprising that phonopy continues to find exactly one vector with a length near zero, and it would be difficult to write a test case where it did fail)